pytorch_lightning/core/lightning.py

```diff
@@ -345,18 +357,35 @@ def log(
                 " but it should not contain information about `dataloader_idx`"
             )
 
-        sync_fn = partial(
-            self.__sync,
-            sync_fn=self.trainer.training_type_plugin.reduce,
-            sync_dist=sync_dist,
-            sync_dist_op=sync_dist_op,
-            sync_dist_group=sync_dist_group,
-            device=self.device,
-        )
-        value = apply_to_collection(value, (torch.Tensor, numbers.Number), sync_fn)
+        if metric_attribute is None and isinstance(value, Metric):
+            if self._metric_attributes is None:
+                # compute once
+                self._metric_attributes = {
+                    id(module): name
+                    for name, module in self.named_children() if isinstance(module, Metric)
+                }
+                if not self._metric_attributes:
+                    raise MisconfigurationException(
+                        "Could not find the `LightningModule` attribute for the `torchmetrics.Metric` logged."
```

Comment:
I guess we don't have the choice but to raise an Exception there to enable fault tolerant logging.

Reply:
n00b question: i don't understand this pattern. could you describe the problem being solved with this?

Reply:
We decided to leave out serialization out of these changes and for a future PR.

But the idea is that we need to know which attribute in the `LightningModule` is associated to the `Metric` being logged. This way, when we reload a `ResultCollection.state_dicit()` we can reassign any metric values on the `LightningModule`

Reply:
Removed this from this PR since it's unused